tools/test/test.sh

```diff
@@ -34,6 +34,13 @@ elif [ "$SUITE" == "docs" ]; then
     tools/generate_doc.py
     test -s $RUCIO_CLIENT_API_OUTPUT
     test -s $RUCIO_REST_API_OUTPUT
+    tools/generate_rest_api_doc.py > api_doc.yaml
+
+    echo "Node version:"
+    node -v
+    npx @redocly/openapi-cli lint api_doc.yaml || true
```

Comment:
So, there seems to be a [openapi python validator in the apispec package itself](https://apispec.readthedocs.io/en/latest/api_core.html?highlight=validation#apispec.utils.validate_spec) 
I think we can get rid of node and just stick with this one ;)

Reply:
Okay I had a look into it and I think there are some flaws with the [python apispec validator](https://apispec.readthedocs.io/en/latest/api_core.html?highlight=validation#apispec.utils.validate_spec):
- The function returns after one errror, this makes the development cycle unnecessary complicated
- The output gets generated from the spec object, not the final spec file. The output itself is hard to read
  Example:
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/prance/__init__.py", line 242, in _validate_openapi_spec_validator
    validate_v3_spec(self.specification)
  File "/usr/local/lib/python3.6/site-packages/openapi_spec_validator/shortcuts.py", line 7, in validate
    return validator_callable(spec, spec_url=spec_url)
  File "/usr/local/lib/python3.6/site-packages/openapi_spec_validator/validators.py", line 50, in validate
    raise err
openapi_spec_validator.exceptions.OpenAPIValidationError: 'responses' is a required property

Failed validating 'required' in schema['properties']['paths']['patternProperties']['^\\/']['patternProperties']['^(get|put|post|delete|options|head|patch|trace)$']:
    {'additionalProperties': False,
     'patternProperties': {'^x-': {}},
     'properties': {'callbacks': {'additionalProperties': {'oneOf': [{'$ref': '#/definitions/Callback'},
                                                                     {'$ref': '#/definitions/Reference'}]},
                                  'type': 'object'},
                    'deprecated': {'default': False, 'type': 'boolean'},
                    'description': {'type': 'string'},
                    'externalDocs': {'$ref': '#/definitions/ExternalDocumentation'},
                    'operationId': {'type': 'string'},
                    'parameters': {'items': {'oneOf': [{'$ref': '#/definitions/Parameter'},
                                                       {'$ref': '#/definitions/Reference'}]},
                                   'type': 'array',
                                   'uniqueItems': True},
                    'requestBody': {'oneOf': [{'$ref': '#/definitions/RequestBody'},
                                              {'$ref': '#/definitions/Reference'}]},
                    'responses': {'$ref': '#/definitions/Responses'},
                    'security': {'items': {'$ref': '#/definitions/SecurityRequirement'},
                                 'type': 'array'},
                    'servers': {'items': {'$ref': '#/definitions/Server'},
                                'type': 'array'},
                    'summary': {'type': 'string'},
                    'tags': {'items': {'type': 'string'}, 'type': 'array'}},
     'required': ['responses'],
     'type': 'object'}

On instance['paths']['/accountlimits/local/{account}/{rse}']['post']:
    {}

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/apispec/utils.py", line 70, in validate_spec
    prance.BaseParser(spec_string=json.dumps(spec.to_dict()), **parser_kwargs)
  File "/usr/local/lib/python3.6/site-packages/prance/__init__.py", line 116, in __init__
    self.parse()
  File "/usr/local/lib/python3.6/site-packages/prance/__init__.py", line 146, in parse
    self._validate()
  File "/usr/local/lib/python3.6/site-packages/prance/__init__.py", line 185, in _validate
    validator(parsed)
  File "/usr/local/lib/python3.6/site-packages/prance/__init__.py", line 246, in _validate_openapi_spec_validator
    raise_from(ValidationError, v3_ex)
  File "/usr/local/lib/python3.6/site-packages/prance/util/exceptions.py", line 24, in raise_from
    raise klass(*args) from from_value
prance.ValidationError: ("'responses' is a required property", 'required', deque(['paths', '/accountlimits/local/{account}/{rse}', 'post']), None, [], ['responses'], {}, {'type': 'object', 'required': ['responses'], 'properties': {'tags': {'type': 'array', 'items': {'type': 'string'}}, 'summary': {'type': 'string'}, 'description': {'type': 'string'}, 'externalDocs': {'$ref': '#/definitions/ExternalDocumentation'}, 'operationId': {'type': 'string'}, 'parameters': {'type': 'array', 'items': {'oneOf': [{'$ref': '#/definitions/Parameter'}, {'$ref': '#/definitions/Reference'}]}, 'uniqueItems': True}, 'requestBody': {'oneOf': [{'$ref': '#/definitions/RequestBody'}, {'$ref': '#/definitions/Reference'}]}, 'responses': {'$ref': '#/definitions/Responses'}, 'callbacks': {'type': 'object', 'additionalProperties': {'oneOf': [{'$ref': '#/definitions/Callback'}, {'$ref': '#/definitions/Reference'}]}}, 'deprecated': {'type': 'boolean', 'default': False}, 'security': {'type': 'array', 'items': {'$ref': '#/definitions/SecurityRequirement'}}, 'servers': {'type': 'array', 'items': {'$ref': '#/definitions/Server'}}}, 'patternProperties': {'^x-': {}}, 'additionalProperties': False}, deque(['properties', 'paths', 'patternProperties', '^\\/', 'patternProperties', '^(get|put|post|delete|options|head|patch|trace)$', 'required']), None)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "tools/generate_rest_api_doc.py", line 37, in <module>
    validate_spec(spec)
  File "/usr/local/lib/python3.6/site-packages/apispec/utils.py", line 72, in validate_spec
    raise exceptions.OpenAPIError(*err.args)
apispec.exceptions.OpenAPIError: ("'responses' is a required property", 'required', deque(['paths', '/accountlimits/local/{account}/{rse}', 'post']), None, [], ['responses'], {}, {'type': 'object', 'required': ['responses'], 'properties': {'tags': {'type': 'array', 'items': {'type': 'string'}}, 'summary': {'type': 'string'}, 'description': {'type': 'string'}, 'externalDocs': {'$ref': '#/definitions/ExternalDocumentation'}, 'operationId': {'type': 'string'}, 'parameters': {'type': 'array', 'items': {'oneOf': [{'$ref': '#/definitions/Parameter'}, {'$ref': '#/definitions/Reference'}]}, 'uniqueItems': True}, 'requestBody': {'oneOf': [{'$ref': '#/definitions/RequestBody'}, {'$ref': '#/definitions/Reference'}]}, 'responses': {'$ref': '#/definitions/Responses'}, 'callbacks': {'type': 'object', 'additionalProperties': {'oneOf': [{'$ref': '#/definitions/Callback'}, {'$ref': '#/definitions/Reference'}]}}, 'deprecated': {'type': 'boolean', 'default': False}, 'security': {'type': 'array', 'items': {'$ref': '#/definitions/SecurityRequirement'}}, 'servers': {'type': 'array', 'items': {'$ref': '#/definitions/Server'}}}, 'patternProperties': {'^x-': {}}, 'additionalProperties': False}, deque(['properties', 'paths', 'patternProperties', '^\\/', 'patternProperties', '^(get|put|post|delete|options|head|patch|trace)$', 'required']), None)
```
compared to the openapi-cli one:
```
[4] api_doc.yaml:8:5 at #/paths/~1accountlimits~1local~1{account}~1{rse}/post

The field `responses` must be present on this level.

 6 | /static/{filename}: {}
 7 | /accountlimits/local/{account}/{rse}:
 8 |   post: {}
   |   ^^^^
 9 |   delete: {}
10 | /accountlimits/global/{account}/{rse_expression}:
```

I think the best approach is to use the `@redocly/openapi-cli` package. We could activate the python linter later, however in the current state the error messages won't provide and value.

Reply:
Haha that's a huge difference :P. Okay, let's stick with node